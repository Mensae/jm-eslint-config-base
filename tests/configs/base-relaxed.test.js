/* eslint-env jest */

const baseConfig = require('../../configs/base-relaxed.js');

describe('eslint-config-base-relaxed', () => {
  it('should match the snapshot', () => {
    expect(baseConfig).toMatchSnapshot();
  });
});
