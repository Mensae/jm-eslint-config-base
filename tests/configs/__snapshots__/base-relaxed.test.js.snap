// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eslint-config-base-relaxed should match the snapshot 1`] = `
{
  "env": {
    "es2022": true,
  },
  "parserOptions": {
    "ecmaVersion": "latest",
    "sourceType": "module",
  },
  "plugins": [
    "@getify/proper-arrows",
    "@getify/proper-ternary",
    "@jm/no-optional-call",
    "import",
    "unicorn",
    "eslint-plugin-no-only-tests",
    "jsdoc",
  ],
  "reportUnusedDisableDirectives": true,
  "rules": {
    "@getify/proper-arrows/name": "warn",
    "@getify/proper-arrows/params": [
      "warn",
      {
        "allowed": [
          "e",
          "v",
          "cb",
          "fn",
          "pr",
        ],
        "count": 2,
        "length": 3,
        "unused": "trailing",
      },
    ],
    "@getify/proper-arrows/return": [
      "warn",
      {
        "ternary": 1,
      },
    ],
    "@getify/proper-arrows/this": "off",
    "@getify/proper-arrows/where": "off",
    "@getify/proper-ternary/nested": [
      "warn",
      {
        "depth": 10,
        "else": true,
      },
    ],
    "@getify/proper-ternary/parens": [
      "warn",
      {
        "call": false,
        "object": false,
        "ternary": false,
      },
    ],
    "@getify/proper-ternary/where": "off",
    "@jm/no-optional-call/no-optional-call": "error",
    "accessor-pairs": "off",
    "array-callback-return": "warn",
    "consistent-return": "off",
    "constructor-super": "error",
    "curly": "off",
    "default-case-last": "off",
    "dot-notation": "off",
    "eqeqeq": "off",
    "for-direction": "error",
    "getter-return": "error",
    "grouped-accessor-pairs": "off",
    "import/default": "error",
    "import/export": "error",
    "import/named": [
      "error",
      {
        "commonjs": true,
      },
    ],
    "import/namespace": "error",
    "import/no-absolute-path": "off",
    "import/no-cycle": "off",
    "import/no-deprecated": "warn",
    "import/no-dynamic-require": "off",
    "import/no-empty-named-blocks": "warn",
    "import/no-mutable-exports": "warn",
    "import/no-named-as-default": "error",
    "import/no-named-as-default-member": "error",
    "import/no-relative-packages": "off",
    "import/no-self-import": "off",
    "import/no-unresolved": "error",
    "import/no-unused-modules": "off",
    "import/no-useless-path-segments": "off",
    "import/no-webpack-loader-syntax": "off",
    "jsdoc/check-access": "error",
    "jsdoc/check-alignment": "error",
    "jsdoc/check-examples": "off",
    "jsdoc/check-param-names": "error",
    "jsdoc/check-property-names": "error",
    "jsdoc/check-syntax": "off",
    "jsdoc/check-tag-names": "error",
    "jsdoc/check-types": "warn",
    "jsdoc/check-values": "warn",
    "jsdoc/empty-tags": "warn",
    "jsdoc/implements-on-classes": "warn",
    "jsdoc/informative-docs": "off",
    "jsdoc/match-description": "off",
    "jsdoc/no-bad-blocks": "warn",
    "jsdoc/no-blank-block-descriptions": "off",
    "jsdoc/no-defaults": "off",
    "jsdoc/no-missing-syntax": "off",
    "jsdoc/no-restricted-syntax": "off",
    "jsdoc/no-types": "off",
    "jsdoc/no-undefined-types": "warn",
    "jsdoc/require-description": "off",
    "jsdoc/require-description-complete-sentence": "off",
    "jsdoc/require-example": "off",
    "jsdoc/require-file-overview": "off",
    "jsdoc/require-jsdoc": "warn",
    "jsdoc/require-param": "warn",
    "jsdoc/require-param-description": "off",
    "jsdoc/require-param-name": "warn",
    "jsdoc/require-param-type": "warn",
    "jsdoc/require-property": "warn",
    "jsdoc/require-property-description": "off",
    "jsdoc/require-property-name": "warn",
    "jsdoc/require-property-type": "warn",
    "jsdoc/require-returns": "warn",
    "jsdoc/require-returns-check": "warn",
    "jsdoc/require-returns-description": "off",
    "jsdoc/require-returns-type": "warn",
    "jsdoc/require-throws": "off",
    "jsdoc/require-yields": "warn",
    "jsdoc/require-yields-check": "warn",
    "jsdoc/valid-types": "warn",
    "new-cap": "off",
    "no-alert": "off",
    "no-array-constructor": "off",
    "no-async-promise-executor": "error",
    "no-await-in-loop": "warn",
    "no-caller": "off",
    "no-case-declarations": "error",
    "no-class-assign": "error",
    "no-compare-neg-zero": "error",
    "no-cond-assign": "error",
    "no-confusing-arrow": "off",
    "no-console": "off",
    "no-const-assign": "error",
    "no-constant-binary-expression": "warn",
    "no-constant-condition": "warn",
    "no-constructor-return": "warn",
    "no-control-regex": "error",
    "no-debugger": "error",
    "no-delete-var": "error",
    "no-dupe-args": "error",
    "no-dupe-class-members": "error",
    "no-dupe-else-if": "error",
    "no-dupe-keys": "error",
    "no-duplicate-case": "error",
    "no-duplicate-imports": "off",
    "no-else-return": "off",
    "no-empty": "error",
    "no-empty-character-class": "error",
    "no-empty-pattern": "error",
    "no-eval": "error",
    "no-ex-assign": "error",
    "no-extend-native": "off",
    "no-extra-bind": "off",
    "no-extra-boolean-cast": "error",
    "no-extra-label": "off",
    "no-extra-semi": "off",
    "no-fallthrough": "error",
    "no-func-assign": "error",
    "no-global-assign": "error",
    "no-implied-eval": "error",
    "no-import-assign": "error",
    "no-inner-declarations": "error",
    "no-invalid-regexp": "error",
    "no-irregular-whitespace": "error",
    "no-iterator": "off",
    "no-label-var": "off",
    "no-labels": "off",
    "no-lone-blocks": "off",
    "no-lonely-if": "off",
    "no-loop-func": "off",
    "no-loss-of-precision": "error",
    "no-misleading-character-class": "error",
    "no-mixed-operators": [
      "warn",
      {
        "allowSamePrecedence": false,
        "groups": [
          [
            "%",
            "**",
          ],
          [
            "%",
            "+",
          ],
          [
            "%",
            "-",
          ],
          [
            "%",
            "*",
          ],
          [
            "%",
            "/",
          ],
          [
            "/",
            "*",
          ],
          [
            "&",
            "|",
            "<<",
            ">>",
            ">>>",
          ],
          [
            "==",
            "!=",
            "===",
            "!==",
          ],
          [
            "&&",
            "||",
          ],
        ],
      },
    ],
    "no-mixed-spaces-and-tabs": "off",
    "no-multi-str": "off",
    "no-nested-ternary": "off",
    "no-new": "off",
    "no-new-func": "off",
    "no-new-native-nonconstructor": "warn",
    "no-new-object": "off",
    "no-new-symbol": "error",
    "no-new-wrappers": "off",
    "no-nonoctal-decimal-escape": "error",
    "no-obj-calls": "error",
    "no-octal": "error",
    "no-octal-escape": "off",
    "no-only-tests/no-only-tests": [
      "warn",
      {
        "block": [
          "And",
          "context",
          "describe",
          "Feature",
          "fixture",
          "Given",
          "it",
          "Scenario",
          "serial",
          "tape",
          "test",
          "Then",
          "When",
        ],
        "focus": [
          "only",
        ],
      },
    ],
    "no-promise-executor-return": "warn",
    "no-proto": "off",
    "no-prototype-builtins": "error",
    "no-redeclare": "error",
    "no-regex-spaces": "error",
    "no-restricted-exports": "off",
    "no-restricted-globals": [
      "warn",
      {
        "message": "Use \`Number.isFinite(..)\` instead, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isFinite#difference_between_number.isfinite_and_global_isfinite",
        "name": "isFinite",
      },
      {
        "message": "Use \`Number.isNaN(..)\` instead, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN#difference_between_number.isnan_and_global_isnan",
        "name": "isNaN",
      },
      "addEventListener",
      "blur",
      "close",
      "closed",
      "confirm",
      "defaultStatus",
      "defaultstatus",
      "event",
      "external",
      "find",
      "focus",
      "frameElement",
      "frames",
      "history",
      "innerHeight",
      "innerWidth",
      "length",
      "location",
      "locationbar",
      "menubar",
      "moveBy",
      "moveTo",
      "name",
      "onblur",
      "onerror",
      "onfocus",
      "onload",
      "onresize",
      "onunload",
      "open",
      "opener",
      "opera",
      "outerHeight",
      "outerWidth",
      "pageXOffset",
      "pageYOffset",
      "parent",
      "print",
      "removeEventListener",
      "resizeBy",
      "resizeTo",
      "screen",
      "screenLeft",
      "screenTop",
      "screenX",
      "screenY",
      "scroll",
      "scrollbars",
      "scrollBy",
      "scrollTo",
      "scrollX",
      "scrollY",
      "self",
      "status",
      "statusbar",
      "stop",
      "toolbar",
      "top",
    ],
    "no-restricted-syntax": [
      "warn",
      {
        "message": "Use \`typeof\` rather than \`instanceof Object\`.",
        "selector": "BinaryExpression[operator='instanceof'][right.name='Object']",
      },
    ],
    "no-return-await": "error",
    "no-script-url": "off",
    "no-self-assign": "error",
    "no-self-compare": "warn",
    "no-sequences": "off",
    "no-setter-return": "error",
    "no-shadow": "off",
    "no-shadow-restricted-names": "error",
    "no-sparse-arrays": "error",
    "no-template-curly-in-string": "warn",
    "no-this-before-super": "error",
    "no-throw-literal": "off",
    "no-undef": "error",
    "no-undef-init": "off",
    "no-unexpected-multiline": "error",
    "no-unmodified-loop-condition": "warn",
    "no-unneeded-ternary": "off",
    "no-unreachable": "error",
    "no-unreachable-loop": "error",
    "no-unsafe-finally": "error",
    "no-unsafe-negation": "error",
    "no-unsafe-optional-chaining": "error",
    "no-unused-expressions": "off",
    "no-unused-labels": "error",
    "no-unused-private-class-members": "warn",
    "no-unused-vars": "error",
    "no-use-before-define": "warn",
    "no-useless-backreference": "error",
    "no-useless-call": "off",
    "no-useless-catch": "error",
    "no-useless-computed-key": "off",
    "no-useless-concat": "off",
    "no-useless-constructor": "off",
    "no-useless-escape": "error",
    "no-useless-rename": "off",
    "no-useless-return": "off",
    "no-var": "off",
    "no-with": "error",
    "object-shorthand": "off",
    "operator-assignment": "off",
    "prefer-arrow-callback": "off",
    "prefer-const": "off",
    "prefer-destructuring": "off",
    "prefer-exponentiation-operator": "warn",
    "prefer-numeric-literals": "warn",
    "prefer-object-spread": "off",
    "prefer-promise-reject-errors": "off",
    "prefer-regex-literals": "off",
    "prefer-rest-params": "off",
    "prefer-spread": "off",
    "prefer-template": "off",
    "require-atomic-updates": "off",
    "require-yield": "error",
    "strict": "off",
    "unicorn/better-regex": "error",
    "unicorn/catch-error-name": "error",
    "unicorn/consistent-destructuring": "error",
    "unicorn/custom-error-definition": "error",
    "unicorn/error-message": "error",
    "unicorn/explicit-length-check": "error",
    "unicorn/new-for-builtins": "error",
    "unicorn/no-abusive-eslint-disable": "error",
    "unicorn/no-array-callback-reference": "off",
    "unicorn/no-array-for-each": "off",
    "unicorn/no-array-method-this-argument": "error",
    "unicorn/no-array-push-push": "error",
    "unicorn/no-await-expression-member": "error",
    "unicorn/no-empty-file": "error",
    "unicorn/no-for-loop": "error",
    "unicorn/no-hex-escape": "error",
    "unicorn/no-instanceof-array": "error",
    "unicorn/no-keyword-prefix": "off",
    "unicorn/no-lonely-if": "error",
    "unicorn/no-negated-condition": "error",
    "unicorn/no-nested-ternary": "off",
    "unicorn/no-new-array": "error",
    "unicorn/no-null": "error",
    "unicorn/no-object-as-default-parameter": "error",
    "unicorn/no-static-only-class": "error",
    "unicorn/no-thenable": "error",
    "unicorn/no-this-assignment": "error",
    "unicorn/no-typeof-undefined": "error",
    "unicorn/no-unnecessary-await": "error",
    "unicorn/no-unreadable-array-destructuring": "error",
    "unicorn/no-unreadable-iife": "error",
    "unicorn/no-unsafe-regex": "warn",
    "unicorn/no-unused-properties": "off",
    "unicorn/no-useless-fallback-in-spread": "error",
    "unicorn/no-useless-length-check": "error",
    "unicorn/no-useless-promise-resolve-reject": "error",
    "unicorn/no-useless-spread": "error",
    "unicorn/no-useless-switch-case": "error",
    "unicorn/no-useless-undefined": "error",
    "unicorn/no-zero-fractions": "error",
    "unicorn/prefer-array-find": "warn",
    "unicorn/prefer-array-flat": "warn",
    "unicorn/prefer-array-flat-map": "warn",
    "unicorn/prefer-array-index-of": "warn",
    "unicorn/prefer-array-some": "warn",
    "unicorn/prefer-at": "off",
    "unicorn/prefer-code-point": "warn",
    "unicorn/prefer-date-now": "warn",
    "unicorn/prefer-default-parameters": "warn",
    "unicorn/prefer-export-from": "warn",
    "unicorn/prefer-includes": "warn",
    "unicorn/prefer-logical-operator-over-ternary": "warn",
    "unicorn/prefer-math-trunc": "error",
    "unicorn/prefer-modern-math-apis": "warn",
    "unicorn/prefer-module": "off",
    "unicorn/prefer-native-coercion-functions": "warn",
    "unicorn/prefer-negative-index": "warn",
    "unicorn/prefer-number-properties": "warn",
    "unicorn/prefer-object-from-entries": [
      "warn",
      {
        "functions": [
          "fromPairs",
          "R.fromPairs",
        ],
      },
    ],
    "unicorn/prefer-optional-catch-binding": "warn",
    "unicorn/prefer-prototype-methods": "warn",
    "unicorn/prefer-reflect-apply": "warn",
    "unicorn/prefer-regexp-test": "warn",
    "unicorn/prefer-set-has": "warn",
    "unicorn/prefer-set-size": "warn",
    "unicorn/prefer-spread": "off",
    "unicorn/prefer-string-replace-all": "warn",
    "unicorn/prefer-string-slice": "warn",
    "unicorn/prefer-string-starts-ends-with": "warn",
    "unicorn/prefer-string-trim-start-end": "warn",
    "unicorn/prefer-switch": "warn",
    "unicorn/prefer-ternary": "warn",
    "unicorn/prefer-top-level-await": "warn",
    "unicorn/prefer-type-error": "error",
    "unicorn/relative-url-style": "error",
    "unicorn/require-array-join-separator": "error",
    "unicorn/require-number-to-fixed-digits-argument": "error",
    "unicorn/throw-new-error": "error",
    "use-isnan": "error",
    "valid-typeof": "error",
    "yoda": "off",
  },
}
`;
